src/Compiler/SyntaxTree/WarnScopes.fsi

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler
+
+open FSharp.Compiler.Diagnostics
+open FSharp.Compiler.SyntaxTrivia
+open FSharp.Compiler.Text
+open FSharp.Compiler.UnicodeLexing
+
+module internal WarnScopes =
```

Comment:
Any reason to make it internal?

We should somehow make the scopes information accessible to FCS users, given that the previous implementation is removed from SyntaxTree.

Reply:
The directives are available as trivia (now in `ParsedImplFileInputTrivia` / `ParsedSigFileInputTrivia`), see commit 5 of this PR.
But if more information is useful for tooling, please let me know what I should make available.

Reply:
Helpers like `IsWarnon` and `IsNowarn` are definitely useful.

Reply:
I made `IsWarnon` and `IsNowarn` public.
I would recommend, though, to rather rely on the filtering mechanism of the diagnostics loggers of the compiler (which use these two functions internally).